ci/k8s/deploy-cf-for-k8s.sh

```diff
@@ -17,6 +17,7 @@ CAPI_IMAGE="cloudfoundry/cloud-controller-ng@$(get_image_digest_for_resource cap
 NGINX_IMAGE="cloudfoundry/capi-nginx@$(get_image_digest_for_resource nginx-docker-image)"
 CONTROLLERS_IMAGE="cloudfoundry/cf-api-controllers@$(get_image_digest_for_resource cf-api-controllers-docker-image)"
 REGISTRY_BUDDY_IMAGE="cloudfoundry/cf-api-package-registry-buddy@$(get_image_digest_for_resource registry-buddy-docker-image)"
+BACKUP_METADATA_GENERATOR_IMAGE="cloudfoundry/cloudfoundry/cf-api-backup-metadata-generator@$(get_image_digest_for_resource backup-metadata-docker-image)"
```

Comment:
I think it might be better if all of these images were optional and to only replace them if provided. I'm not sure if there is a way to make "get" resources optional.

Reply:
I would prefer we don't make them optional so that _all_ of our latest changes are always deployed to samus, not just a subset. I feel like we would be missing out on a lot of important feedback if we didn't always use the latest versions of those resources

Reply:
I think Samus should already take them all. But the env that runs the backup metadata tests ideally can just use what's on `ci-passed` for the other images, you know? Since the deploy script wants everything I have to override them.

Once we merge the `backup-metadata` pipeline into the one it won't be that big of a deal, though. Right now it just means I have to declare a bunch of image resources that may or may not have actually made it through `samus` in that pipeline.